File: AliyunPlayerDemo/AliyunPlayerMediaDemo/Advanced player/lists_scroll/AliyunPlaySDKDemoFullScreenScrollCollectionViewCell.m
//
//  AliyunPlaySDKDemoFullScreenScrollCollectionViewCell.m
//  AliyunPlayerMediaDemo
//
//  Created by 王凯 on 2017/8/16.
//  Copyright © 2017年 com.alibaba.ALPlayerVodSDK. All rights reserved.
//

#import "AliyunPlaySDKDemoFullScreenScrollCollectionViewCell.h"
#import "Until.h"
#import "UIView+Layout.h"

@interface AliyunPlaySDKDemoFullScreenScrollCollectionViewCell()

@end
@implementation AliyunPlaySDKDemoFullScreenScrollCollectionViewCell
- (UIView *)playerView{
    if(!_playerView){
        _playerView = [[UIView alloc] init];
    }
    return _playerView;
}


- (instancetype)initWithFrame:(CGRect)frame{
    if (self = [super initWithFrame:frame]) {
        self.backgroundColor = [UIColor grayColor];
        self.playerView.frame= CGRectMake(10, 10, SCREEN_WIDTH-20, SCREEN_HEIGHT-20);
        self.playerView.backgroundColor = [UIColor blackColor];
        [self addSubview:self.playerView];
    }
    return self;
}
-(void)layoutSubviews{
    
}


@end

File: AliyunPlayerDemo/AliyunPlayerMediaDemo-Swift/多实例滑动/AliyunPlaySDKDemoFullScreenScrollCollectionViewCell.swift
//
//  AliyunPlaySDKDemoFullScreenScrollCollectionViewCell.swift
//  AliyunPlayerMediaDemo-Swift
//
//  Created by baby on 2017/10/14.
//  Copyright © 2017年 com.alibaba.ALPlayerVodSDK. All rights reserved.
//

import UIKit

class AliyunPlaySDKDemoFullScreenScrollCollectionViewCell: UICollectionViewCell {
    
    lazy var aliyunVodPlayer:AliyunVodPlayer = {
        //播放器初始化
        let tempPlayer = AliyunVodPlayer()
        tempPlayer.isAutoPlay = true
        return tempPlayer
    }()
    
    override init(frame: CGRect) {
        super.init(frame:frame)
        
        let playView = self.aliyunVodPlayer.playerView
        playView?.frame = CGRect(x: 0, y: 0, width: SCREEN_WIDTH, height: SCREEN_HEIGHT)
        self.contentView.addSubview(playView!)
        
    }
    
    
    required init?(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
    
    deinit {
        aliyunVodPlayer.stop()
        aliyunVodPlayer.release()
    }
    
}
